apps/opik-backend/src/test/java/com/comet/opik/api/resources/utils/resources/PromptResourceClient.java

```diff
@@ -0,0 +1,65 @@
+package com.comet.opik.api.resources.utils.resources;
+
+import com.comet.opik.api.CreatePromptVersion;
+import com.comet.opik.api.Prompt;
+import com.comet.opik.api.PromptVersion;
+import com.comet.opik.api.resources.utils.TestUtils;
+import com.comet.opik.infrastructure.auth.RequestContext;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.HttpHeaders;
+import org.apache.hc.core5.http.HttpStatus;
+import ru.vyarus.dropwizard.guice.test.ClientSupport;
+import uk.co.jemos.podam.api.PodamFactory;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public interface PromptResourceClient {
+
+    String PROMPT_PATH = "%s/v1/private/prompts";
+
+    record PromptResourceClientParams(ClientSupport client, String baseURI, PodamFactory podamFactory) {
+    }
+
+    static PromptResourceClient create(ClientSupport client, String baseURI, PodamFactory podamFactory) {
+        return () -> new PromptResourceClientParams(client, baseURI, podamFactory);
+    }
+
+    PromptResourceClientParams params();
```

Comment:
I guess it's not readable and an overkill for a simple rest call with payload. Also you are creating new object with a set of params for every call. 
Can we just make a regular object with private params, and initialize it in `@BeforeAll`, to be used by all tests. It will be much easier later to read(which is important) and will save some memory(minor).

Reply:
This builder pattern is much more readable from the user's perspective, but the object will add more boilerplate. That's okay, and the reusing it depends on where the client instantiates it. Memory would be insignificant since these would be only references and quickly collected by the garbage collector.